### PR TITLE
fix(vault): Address Vault related issues

### LINKED ISSUE #324: V2: Failed to initialize Credential manager
User may be forced to use plain text credentials.

image

---

Whenever there is a populated `secure: [ ... ]` array in a zowe.config.json file, the CICS extension fails to load V2 profiles.
<details><summary>Example `zowe.config.json` with secure properties</summary>
<p>

```jsonc
// zowe.config.json
{
    "$schema": "./zowe.schema.json",
    "profiles": {
        "zosmf": {
            "type": "zosmf",
            "properties": {
                "port": 443
            }
        },
        "cics": {
            "type": "cics",
            "properties": {
                "port": 5555,
                "protocol": "https",
                "regionName": "REGION"
            }
        },
        "base": {
            "type": "base",
            "properties": {
                "host": "my.host.net",
                "rejectUnauthorized": false,
                "port": 1234
            },
            "secure": [
                "user",
                "password"
            ]
        },
    },
    "defaults": {
        "zosmf": "zosmf",
        "cics": "cics",
        "base": "base"
    },
    "autoStore": true
}

//~/.zowe/settings/imperative.json
{
  "overrides": {
    "CredentialManager": "@zowe/cli"
  }
}
```

</p>
</details> 

<details><summary>Equivalent `zowe.config.json` without secure properties</summary>
<p>

```jsonc
// zowe.config.json
{
    "$schema": "./zowe.schema.json",
    "profiles": {
        "zosmf": {
            "type": "zosmf",
            "properties": {
                "port": 443
            }
        },
        "cics": {
            "type": "cics",
            "properties": {
                "port": 5555,
                "protocol": "https",
                "regionName": "REGION"
            }
        },
        "base": {
            "type": "base",
            "properties": {
                "host": "my.host.net",
                "rejectUnauthorized": false,
                "port": 1234,
                "user": "my-user",
                "password": "my-password"
            }
        },
    },
    "defaults": {
        "zosmf": "zosmf",
        "cics": "cics",
        "base": "base"
    },
    "autoStore": true
}

//~/.zowe/settings/imperative.json
{
  "overrides": {
    "CredentialManager": false
  }
}
```

</p>
</details> 

---

The following PR hides the error on refresh:
- #326 

However, the problem still persist since no V2 profiles get loaded, and you can get the error to show again after doing a `+` (_Create a CICS profile_) operation

### PR DESCRIPTION
**Pull request by zFernand0**
In this PR:
- Fix #324
- Updated the workflows to be ready to publish via `workfow_dispatch`
- Update `.gitignore` to exclude some dev-generated files
- Added the `.env` file to the `launch.json` configuration
- Removed `.md` files from the `update.licenses.js` script

TODO:
- [x] Update the `@zowe/zowe-explorer-api` dependency once the following PRs are merged (and published)
  - https://github.com/zowe/vscode-extension-for-zowe/pull/2397
  - https://github.com/zowe/vscode-extension-for-zowe/pull/2423
  - https://github.com/zowe/vscode-extension-for-zowe/pull/2426

Note:
- You may run into #329
  - I worked around it by installing the plugin with the cli and running `zowe config schema > ~/.zowe/zowe.schema.json`
  - Another workaround is to run the `Initialize Team Configuration File` (a.k.a. `zowe.all.config.init`) command from the command palette
    - This may show the prompt below... Just hit the "Create New" button
      image

### CODE REVIEW

**Comment by zFernand0** [comment]
After this is merged, I will kickoff the workflow to publish a new 2.3.1 of CICS. 😋

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,34 +1,99 @@
...
         run: npm i
 
+      - name: Build
+        run: npm run build
+
+      - name: Lint
+        run: npm run lint
+
+      # - name: Tests
+      #   run: npm run test
```
Comment (t1m0thyj) [inline .github/workflows/main.yml:50]: Curious why the test stage is commented out 😋

**Review by awharn — APPROVED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -1,34 +1,99 @@
...
+
+  deploy_vsix:
```
Comment (t1m0thyj) [inline .github/workflows/main.yml:52]: Should this job have a "needs" clause so it waits for the `build_vsix` job to succeed?

We may need to re-enable the workflow for further testing 😋

**Review by JillieBeanSim — APPROVED** [review]
LGTM thanks for your work on this @zFernand0

**Review by t1m0thyj — APPROVED** [review]
LGTM, left a few questions about the workflows 🙂

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,34 +1,99 @@
...
         with:
-          node-version: 14
-          registry-url: https://registry.npmjs.org/
+          node-version: ${{ matrix.node-version }}
 
       - name: Install Dependencies
         run: npm i
 
+      - name: Build
+        run: npm run build
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply .github/workflows/main.yml:44]: working builds
- Windows: node 18: https://github.com/zowe/vscode-extension-for-cics/actions/runs/6000721117/job/16273400213
- Windows: node 16: https://github.com/zowe/vscode-extension-for-cics/actions/runs/6000721117/job/16273399388

**Comment by zFernand0** [comment]
We should be ok to merge this PR, even though the `deploy` step will run (and likely fail).
The deployment is intended to publish the extension when triggered manually from the Actions tab 😋 
After a 2.3.1, we can disable the workflow again! 😋

**Pull request merged**